### PR TITLE
Adding "autoRemove" field for HostConfiguration.

### PR DESCRIPTION
**Pull request by diemol**
Hi, 

Regarding the discussion related to this [question](https://github.com/spotify/docker-client/issues/488) I had some weeks ago, here is the PR addressing that field I was asking for and that will be added in the [API v.1.25](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.25.md).

I added also a small unit test to check the field.

### CODE REVIEW

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -3356,6 +3356,26 @@ public void testOomScoreAdj() throws Exception {
...
+        .image(BUSYBOX_LATEST)
+        .hostConfig(HostConfig.builder()
+                        .autoRemove(true) // Default is false
+                        .build())
+        .build();
+
+    final ContainerCreation container = sut.createContainer(config, randomName());
+    final ContainerInfo info = sut.inspectContainer(container.id());
+
+    assertThat(info.hostConfig().autoRemove(), is(true));
```
Comment (johnflavin) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: In this test, you are just setting the option on container creation and verifying that it was indeed set correctly. A better test would be to start and stop the container, verifying that it does get removed.

Comment (diemol) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: You are right, the test I wrote was quite naive. Anyway, I improved it a bit and just made the commit.
Thanks for the feedback.

**Comment by diemol** [comment]
Hi @mattnworb, 

To test it locally I just faked the behaviour as if the flag was implemented. But I think this does not make sense in the CI and overall, honestly, I would just wait until it gets released.

Right now I have a workaround for that missing feature, so I don't need it urgently, it is more a nice to have.

Thanks for the nice work you are doing over there, this docker-client simplified my life a lot in the project I am working, so the least I could do is to five something back :smile:

File: src/main/java/com/spotify/docker/client/messages/HostConfig.java [hunk 2]
```diff
@@ -1042,6 +1053,15 @@ public Integer oomScoreAdj() {
       return oomScoreAdj;
     }
 
+    public Boolean autoRemove() {
+      return autoRemove;
+    }
+
+    public Builder autoRemove(final Boolean autoRemove) {
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: On second thought I am happy to merge this now, in advance of the 1.25 API being released, but would you mind adding a javadoc comment here that this field only has relevance for versions >= 1.25?

Comment (diemol) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: Sure, I'll do it between today and tomorrow!

**Comment by davidxia** [comment]
I've added the javadoc here https://github.com/spotify/docker-client/pull/538

**Comment by diemol** [comment]
Ah cool @davidxia, you were faster! Thanks!

**Pull request closed without merging**